bin/varnishd/mgt/mgt_jail_linux.c

```diff
@@ -48,11 +48,61 @@
 #include "mgt/mgt.h"
 #include "common/heritage.h"
 
+static int
+vjl_set_thp(const char *arg)
+{
+	int val;
+
+	if (!strcmp(arg, "ignore"))
+		return (0);
+	if (!strcmp(arg, "enable"))
+		val = 0;
+	else if (!strcmp(arg, "disable"))
+		val = 1;
+	else {
+		ARGV_ERR(
+		    "linux jail: unknown value '%s' for argument transparent_hugepage.\n",
+		    arg);
+	}
+	if (prctl(PR_SET_THP_DISABLE, val, 0, 0, 0) != 0) {
+		MGT_Complain(C_ERR,
+		    "Could not %s Transparent Hugepage: %s (%d)",
+		    arg, VAS_errtxt(errno), errno);
+	}
+	return (0);
+}
+
 static int
 vjl_init(char **args)
 {
+	char **unix_args;
+	int ret = 0;
+	size_t i;
+
+	if (args == NULL) {
+		/* Autoconfig */
+		if (vjl_set_thp("disable") != 0)
+			return (1);
+		return (jail_tech_unix.init(NULL));
+	}
 
-	return jail_tech_unix.init(args);
+	for (i = 0; args[i] != NULL; i++);
+	unix_args = calloc(i + 1, sizeof *unix_args);
+	AN(unix_args);
+
+	for (i = 0; *args != NULL && ret == 0; args++) {
+		if (!strncmp(*args, "transparent_hugepage=", 22)) {
+			ret = vjl_set_thp((*args) + 22);
```

Comment:
The offset is 21, because that is half 42.

Also, this is a mouthful, maybe we should call the option `huge` (like the mount option) instead of the verbose `transparent_hugepage`.

Opinions?

Reply:
I feel like having a more precise name like currently makes it less confusing than using simply `huge`